src/Raven.Server/Documents/PeriodicBackup/RavenFtpClient.cs

```diff
@@ -65,135 +67,121 @@ public void UploadFile(string folderName, string fileName, Stream stream)
 
             // create the nested folders including the new folder
             directories.Add(folderName);
+            string url = "";
 
             foreach (var directory in directories)
             {
                 if (directory == string.Empty)
                     continue;
 
                 url += $"/{directory}";
-                var request = CreateFtpWebRequest(url, WebRequestMethods.Ftp.MakeDirectory, keepAlive: false);
-
-                try
-                {
-                    var response = request.GetResponse();
-                    response.Close();
-                }
-                catch (WebException e)
+                using (var client = CreateFtpClient(_url, keepAlive: false))
                 {
-                    var response = (FtpWebResponse)e.Response;
-                    if (response.StatusCode == FtpStatusCode.ActionNotTakenFileUnavailable)
+                    try
                     {
-                        // folder already exists
-                        continue;
+                        if (!client.DirectoryExists($"/{directory}"))
+                            client.CreateDirectory($"/{directory}");
+                    }
+                    catch (WebException e)
+                    {
+                        throw ;
                     }
-
-                    throw;
                 }
             }
 
             return url;
         }
 
-        private void ExtractUrlAndDirectories(out string url, out List<string> dirs)
+        private void ExtractUrlAndDirectories(out List<string> dirs)
         {
             var uri = new Uri(_url);
 
             dirs = uri.AbsolutePath.TrimStart('/').TrimEnd('/').Split("/").ToList();
-            var port = _port ?? (uri.Port > 0 ? uri.Port : DefaultFtpPort);
-            if (port < 1 || port > 65535)
-                throw new ArgumentException("Port number range: 1-65535");
-
-            url = $"{uri.Scheme}://{uri.Host}:{port}";
         }
 
-        private FtpWebRequest CreateFtpWebRequest(string url, string method, bool keepAlive)
+        internal FtpClient CreateFtpClient(string url, bool keepAlive)
         {
-            var request = (FtpWebRequest)WebRequest.Create(new Uri(url));
-            request.Method = method;
-
-            request.Credentials = new NetworkCredential(_userName, _password);
-            request.EnableSsl = _useSsl;
+            var client = new FtpClient(url);
+            client.Config.ConnectTimeout = -1;
+            client.Credentials = new NetworkCredential(_userName, _password);
             if (_useSsl)
             {
                 var byteArray = Convert.FromBase64String(_certificateAsBase64);
 
                 try
                 {
-                    var x509Certificate = new X509Certificate(byteArray);
-                    request.ClientCertificates = new X509CertificateCollection(new[] { x509Certificate });
+                    var x509Certificate = new X509Certificate2(byteArray);
```

Comment:
it would be great to have a test for this case

Reply:
Done